src/Symfony/Component/DependencyInjection/Loader/Tag/IteratorTag.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Loader\Tag;
+
+use Symfony\Component\DependencyInjection\Argument\IteratorArgument;
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+use Symfony\Component\Yaml\Tag\TagInterface;
+use Symfony\Component\Yaml\Tag\DumpableTagInterface;
+
+/**
+ * @author Guilhem N. <egetick@gmail.com>
+ *
+ * @internal
+ */
+final class IteratorTag implements TagInterface, DumpableTagInterface
```

Comment:
there is also "=closure_proxy" which could benefit from yaml tags.
which means we could have only one class - eg "ArgumentTag" for the DI component, able to deal with all tags it'll need to support (these 2 today and more in the future, no forward compat issue here)